modules/acm/main.tf

```diff
@@ -37,7 +50,22 @@ resource "aws_acm_certificate" "this" {
 # ACM Validation
 ################################################################################
 
-resource "aws_route53_record" "this" {
+resource "aws_route53_record" "same_account" {
+  count = var.route53_assume_role_arn == null ? 1 : 0
+
+  zone_id         = var.record_zone_id
+  name            = local.acm_certificate_validation_record.name
+  type            = local.acm_certificate_validation_record.type
+  records         = [local.acm_certificate_validation_record.value]
+  ttl             = 60
+  allow_overwrite = var.record_allow_overwrite
+}
+
+resource "aws_route53_record" "cross_account" {
```

Comment:
let's have the name as something more descriptive than same account and cross account. 
also wouldn't changing name of existing resource require a destroy and create? We should avoid it in that case and only use a different name for the new `aws_route53_record` resource being added.